gnucash/import-export/import-main-matcher.cpp

```diff
@@ -2028,6 +2043,22 @@ refresh_model_row (GNCImportMainMatcher *gui,
                                      imbalance, acct_full_name);
 
                 }
+                if (auto recn_date = acct_recndate_warning (date, dest_acc))
+                {
+                    static const char* recn_date_warning = N_("The import date %s is \
+earlier than destination account %s whose last reconciled date is %s. Further \
+reconciliation of the destination account may be difficult.");
```

Comment:
I think you want to say `"The import includes at least one transaction whose posted date %s is before the last reconciliation date %s of account %s for which it has a split. This may make reconciliation of that account difficult."

But that's utter nonsense. GnuCash's reconciliation mechanism handles that situation very nicely. It's a pretty common occurence and raising a warning about it will only be annoying.

Reply:
I'd offer an alternative view... it's common to have multiple accounts, eg checking, savings, spouse bank accounts tracked.

If I have reconciled my checking account with the bank statement dated 31-dec-2023 successfully, it means all bank's transactions have been matched with my book account. It would be _unusual_ that I'd want to add another split in my checking account _before_ 31-dec-2023. In my view, any import which automatically or manually matches to the checking account before 31-dec-2023 is a mismatch, and it would be nice to detect this.

Reply:
In my usage it's quite common for a payment I posted not to clear the bank for a few days, sometimes even weeks and in that case the transaction won't get reconciled until a later statement. It would be pretty annoying for GnuCash to pop a warning dialog every time that happens. It certainly doesn't make reconciling more difficult, the older transactions keep showing up in the reconcile window until they're reconciled, and you don't reconcile them until they show up on the bank statement. 

Of course that doesn't work if you only enter transactions by importing them from the bank, but if you do that you're keeping the bank's books not your own and you have no way of noticing when a transaction doesn't clear. There's not really any reconciliation going on either: You're just downloading transactions from the bank and marking them reconciled. Since both the downloaded transactions and the statement come from the same database if not the same computer it would be pretty strange if they didn't agree.

Reply:
This warning isn't about handling unreconciled splits in an account already reconciled. It's about popping up a soft warning that we're adding a new split in a reconciled account via import. The warning will only pop up as a yellow triangle instead of the red/yellow/green bars, which is not intrusive.

Reply:
In my scenario I will exercise much care in _manually_ entering data in my main checking account: it will receive all paychecks, it is regularly used to pay various operating expenses. It is also often reconciled with the bank statement.

I also hold other accounts, eg savings accounts. I'd rely on the bank imports.

This work implements a soft warning only, ensuring the main checking account doesn't receive new transactions before the reconciliation date.

Reply:
> The warning will only pop up as a yellow triangle instead of the red/yellow/green bars, which is not intrusive.

 Not exactly, at least according to your screenshot. You add a new column that displays an icon if there's a warning on the transaction. (Oddly the example is warning about an expense account. Equity accounts normally don't represent an external account to reconcile with.) It's not as intrusive as the dialog box in your first pass but it's still intrusive.

> This work implements a soft warning only, ensuring the main checking account doesn't receive new transactions before the reconciliation date.

Ensuring nothing. It adds a flag with a misleading tooltip attached when an import adds a transaction to the import account when the *matched* account has a reconciled date after the transaction's post date.

How might that happen? Either the matcher selected the wrong "other" account or the transaction exists but the matcher failed to match it or the transaction exists  in the "other" account but with a different other account than the one you're importing now. Relying on the "other" account's last reconciled date to catch any of those is dicey: It depends on the reconcile periods of the various accounts being the same and on the order that the accounts are reconciled. It's more reliable to review carefully the import in the matcher window before clicking OK. In your case any added transaction with the checking account as a match is a problem regardless of reconciliation date. If you've already done savings account A's import and it turns up as a matched account in an add in savings account B's import that's a problem too.

Reply:
> > The warning will only pop up as a yellow triangle instead of the red/yellow/green bars, which is not intrusive.
> 
> Not exactly, at least according to your screenshot. You add a new column that displays an icon if there's a warning on the transaction. (Oddly the example is warning about an expense account. Equity accounts normally don't represent an external account to reconcile with.) It's not as intrusive as the dialog box in your first pass but it's still intrusive.

This was a contrived example. There's no additional column. It reuses the existing automatch pixbuf column.

A better example with standard icon:

![image](https://github.com/Gnucash/gnucash/assets/1975870/07b19b41-92ff-4994-b6ad-1cc51e120883)

> > This work implements a soft warning only, ensuring the main checking account doesn't receive new transactions before the reconciliation date.
> 
> Ensuring nothing. It adds a flag with a misleading tooltip attached when an import adds a transaction to the import account when the _matched_ account has a reconciled date after the transaction's post date.

As above it won't prevent the user from adding the transaction, but will flag that the account is _probably_ an incorrect auto-match or manual-match.

> How might that happen? Either the matcher selected the wrong "other" account or the transaction exists but the matcher failed to match it or the transaction exists in the "other" account but with a different other account than the one you're importing now. Relying on the "other" account's last reconciled date to catch any of those is dicey: It depends on the reconcile periods of the various accounts being the same and on the order that the accounts are reconciled.

> It's more reliable to review carefully the import in the matcher window before clicking OK. 

Exactly^ it would be nice to flag automatically these imports whereby an algorithm can warn of obvious warnings that the user _may_ miss. See similar in assistant-stock-transaction.cpp -- it flags, but doesn't prevent adding a new transaction dated _before_ the stock account _latest_ split, which could affect capgains calculations.